### DISCUSSION TITLE
fallback languague always gets loaded

### DISCUSSION THREAD

**Discussion by BennyAlex** [i18next/i18next#2035] [Q&A] [answered]
<img width="402" alt="image" src="https://github.com/i18next/i18next/assets/17402724/d4fb6ab1-2bb6-425f-a6a4-3c38a3fc4b68">

Why does the fallback language always gets loaded, even tho both have the same keys?
I also don't see any warnings in the console that a key is missing.

I am using react and I am waiting for the 'ready' property (dont use suspense).  Is this the problem? 
If I set fallbackLng to false it only loads the current language, but then I get problems if the detected language is not available. Here is my config:

```

i18next
  // load translation using http -> see /public/locales (i.e. https://github.com/i18next/react-i18next/tree/master/example/react/public/locales)
  // learn more: https://github.com/i18next/i18next-http-backend
  .use(Backend)
  // detect user language
  // learn more: https://github.com/i18next/i18next-browser-languageDetector
  .use(LanguageDetector)
  // pass the i18n instance to react-i18next.
  .use(initReactI18next)
  // init i18next
  // for all options read: https://www.i18next.com/overview/configuration-options
  .init(
    {
      fallbackLng: 'en',
      //fallbackLng: false,
      debug: false,
      useSuspense: false,

      saveMissing: true,

      missingKeyHandler: (lngs, ns, key, fallbackValue) => {
        // Handle the missing key, for example, log it to the console

        console.warn(
          `Missing translation for key: "${key}" in namespace: "${ns}" for languages: "${lngs.join(
            ', '
          )}". Fallback value: "${fallbackValue}"`
        );
      },

      interpolation: {
        escapeValue: false // not needed for react as it escapes by default
      },
      // react-i18next options
      react: {
        useSuspense: false,
        bindI18n: 'languageChanged',
      },
      backend: {
        loadPath: '/locales/{{lng}}/{{ns}}.json'
      },
      load: 'languageOnly' // only load main language code, not country specific

    }
  );
```

**Comment by jamuhl** [comment]
That's how it works...fallback gets always loaded...as it does not know what is available until it actually loaded the files

**Reply by BennyAlex** [reply]
It would also be possible to fetch the current language file First, and only if a key is missing than the fallback language Could be loaded.

Would also be a good Info in the documentation.

**Reply by jamuhl** [reply, marked as answer]
you can't fetch the file of fallback language inside the t function upon being a missing key -> i18next.t run sync why fetching translations is async 

there are three options:

a) you load and use fallbacks

b) you assert resource files are complete (not that hard) and set fallback to false

c) you load the translations for fallback in the missingKeyHandler (still sync and not async)

regarding extending the docs...contributions are always welcome to improve the docs
